geowebcache/wmts/src/main/java/org/geowebcache/service/wmts/WMTSGetCapabilities.java

```diff
@@ -128,8 +157,119 @@ private String generateGetCapabilities(Charset encoding) {
         }
     }
 
-    private void serviceIdentification(XMLBuilder xml) throws IOException {
+    /**
+     * Composes service information using information provided by extensions.
+     */
+    private ServiceInformation getServiceInformation() {
         ServiceInformation servInfo = tld.getServiceInformation();
+        for (WMTSExtension extension : extensions) {
+            ServiceInformation serviceInformation = extension.getServiceInformation();
+            if (serviceInformation != null) {
+                if (servInfo == null) {
+                    servInfo = new ServiceInformation();
+                }
+                mergeServiceInformation(servInfo, serviceInformation);
+            }
+        }
+        return servInfo;
+    }
+
+    /**
+     * Substitute serviceA information with no NULL information from serviceB.
+     */
+    private void mergeServiceInformation(ServiceInformation serviceA, ServiceInformation serviceB) {
```

Comment:
This makes the code order dependent, wondering if we should add a GeoServer ExtensionPriority like interface to GWC as well?